chain/default_syncer.go

```diff
@@ -183,6 +183,13 @@ func (syncer *DefaultSyncer) tipSetState(ctx context.Context, tsKey string) (sta
 // Precondition: the caller of syncOne must hold the syncer's lock (syncer.mu) to
 // ensure head is not modified by another goroutine during run.
 func (syncer *DefaultSyncer) syncOne(ctx context.Context, parent, next types.TipSet) error {
+	head := syncer.chainStore.Head()
+
+	// if tipset is already head, we've been here before. do nothing.
+	if head.Equals(next) {
```

Comment:
Added this check to avoid errors when processing the same tipset twice. This is more likely after moving the lock after the chain fetch.